functions/controllers/announcements-api/index.js

```diff
@@ -0,0 +1,87 @@
+const functions = require("firebase-functions");
+const helmet = require("helmet");
+const express = require("express");
+const cors = require("cors");
+const { corsConfig } = require("../../utils/config");
+const { jwtCheck, validKey, hasDeleteAnnouncement, hasUpdateAnnouncement } = require("../../utils/middleware");
+const { addDocument, queryCollectionSorted, deleteDocument } = require("../../utils/database");
+
+const announcements = express();
+const corsOptions = {
+  origin: corsConfig,
+  optionsSuccessStatus: 200,
+};
+
+announcements.disable("x-powered-by");
+announcements.use(helmet());
+announcements.use(cors(corsOptions));
+announcements.use(express.json());
+
+announcements.get("/", validKey, async (req, res) => {
```

Comment:
For this part, do you think we might need an additional middleware for read permissions as well?

Reply:
So we don't know if this will eventually go on the portal or on the main page. This is assuming only if this is on the main page. If it's in the portal view we can add that permission. In 2020 it was in the portal but it's early to say because realistically even non-registered hackers could view this without.